src/lib/bootloader/bootloader_base.rb

```diff
@@ -9,16 +10,38 @@
 module Bootloader
   # Represents base for all kinds of bootloaders
   class BootloaderBase
+    include Yast::I18n
+
     def initialize
+      textdomain "bootloader"
+
       @read = false
       @proposed = false
+      @initial_sysconfig = Sysconfig.from_system
+    end
+
+    # Prepares the system to (before write the configuration)
+    #
+    # Writes the new sysconfig and, when the Mode.normal is set, tries to install the required packages.
+    # If user decides to cancel the installation, it restores the previous sysconfig.
+    #
+    # @return [Boolean] true whether the system could be prepared as expected;
+    #                   false when user cancel the installation of needed packages
+    def prepare
+      write_sysconfig
+
+      return true unless Yast::Mode.normal
+      return true if Yast::PackageSystem.InstallAll(packages)
+
+      restore_initial_sysconfig
+
+      Yast2::Popup.show(_("Cannot continue without install required packages"))
+      false
     end
 
     # writes configuration to target disk
     def write
-      write_sysconfig
-      # in running system install package, for other modes, it need specific handling
-      Yast::PackageSystem.InstallAll(packages) if Yast::Mode.normal
+      raise NotImplementedError
```

Comment:
this cannot work. All children call super, so it raise exception. If there is nothing to do, just keep it empty.

Reply:
EmptyBootloader will still fail.

Reply:
I remove from children, but I will to leave it empty.